### PR TITLE
Add missing Clear() to some Crypto primitives

### PR DESCRIPTION
**Pull request by danmosemsft**
Fixes https://github.com/dotnet/corefx/issues/11132

@steveharter

### CODE REVIEW

**Comment by danmosemsft** [comment]
@dotnet-bot test this

**Comment by danmosemsft** [comment]
@dotnet-bot test Innerloop OSX Debug Build and Test (`/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx_debug_prtest/layout/layout.proj(74,5): error MSB3030: Could not copy the file "/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx_debug_prtest/bin/Windows_NT.x64.Debug/Native_aot/clrcompression.dll" because it was not found.` .. will ask wes)

File: src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs [hunk 1]
```diff
@@ -92,6 +92,11 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        public void Clear() 
+        {
+            (this as IDisposable).Dispose();
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs]: There's a subtle difference between ```this.Dispose()``` and ```(this as IDisposable).Dispose()```... if a derived type re-implements the IDisposable interface explicitly, the latter will end up using it.  I assume that's the desired behavior?  Just verifying.

File: src/System.Security.Cryptography.Primitives/tests/CryptoStream.cs [hunk 2]
```diff
@@ -159,6 +159,18 @@ public static void NestedCryptoStreams()
         }
 
         [Fact]
```
Comment (steveharter) [inline src/System.Security.Cryptography.Primitives/tests/CryptoStream.cs]: You'll need the #if netcoreapp11 around these tests

File: src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HashAlgorithm.cs]: It's what [Desktop did](http://index/?query=hashalgorithm&rightProject=mscorlib&file=system%5Csecurity%5Ccryptography%5Chashalgorithm.cs&line=206) although perhaps for no particular reason.

File: src/System.Security.Cryptography.Primitives/tests/CryptoStream.cs [hunk 3]
```diff
@@ -158,6 +158,20 @@ public static void NestedCryptoStreams()
...
+#if netcoreapp11 
+        [Fact]
+        public static void Clear()
+        {
+            ICryptoTransform encryptor = new IdentityTransform(1, 1, true);
+            using (MemoryStream output = new MemoryStream())            
+            using (CryptoStream encryptStream = new CryptoStream(output, encryptor, CryptoStreamMode.Write))
+            {
+                encryptStream.Clear();
+                Assert.Throws<NotSupportedException>(() => encryptStream.Write(new byte[] { 1, 2, 3, 4, 5 }, 0, 5));
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Primitives/tests/CryptoStream.cs]: I totally would not have expected this behavior, but I see that it's what desktop would have done.  Good test, good test.

**Pull request merged**